src/overlays/actors/ovl_Demo_Effect/z_demo_effect.c

```diff
@@ -20,121 +104,2126 @@ const ActorInit Demo_Effect_InitVars = {
+            this->envOpaColor[2] = 0x00;
+            break;
+    }
+}
+
+/**
+ * Sets the Jewel color based on the alpha variable.
+ * This function if a value of less than 1.0f is supplied will drain the color from the Jewels.
+ * This effect can be seen in prerelease screenshots.
+ */
+void DemoEffect_SetJewelColor(DemoEffect* this, f32 alpha) {
+    DemoEffect_InitJewelColor(this);
+
+    // s32 casts needed to prevent u8s from casting to float
+    this->primXluColor[0] = (((s32)this->primXluColor[0]) * alpha) + (255.0f * (1.0f - alpha));
+    this->primXluColor[1] = (((s32)this->primXluColor[1]) * alpha) + (255.0f * (1.0f - alpha));
+    this->primXluColor[2] = (((s32)this->primXluColor[2]) * alpha) + (255.0f * (1.0f - alpha));
+    this->primOpaColor[0] = (((s32)this->primOpaColor[0]) * alpha) + (255.0f * (1.0f - alpha));
+    this->primOpaColor[1] = (((s32)this->primOpaColor[1]) * alpha) + (255.0f * (1.0f - alpha));
+    this->primOpaColor[2] = (((s32)this->primOpaColor[2]) * alpha) + (255.0f * (1.0f - alpha));
+    this->envXluColor[0] = ((s32)this->envXluColor[0]) * alpha;
+    this->envXluColor[1] = ((s32)this->envXluColor[1]) * alpha;
+    this->envXluColor[2] = ((s32)this->envXluColor[2]) * alpha;
+    this->envOpaColor[0] = ((s32)this->envOpaColor[0]) * alpha;
+    this->envOpaColor[1] = ((s32)this->envOpaColor[1]) * alpha;
+    this->envOpaColor[2] = ((s32)this->envOpaColor[2]) * alpha;
+}
+
+/**
+ * Moves the Jewel Actor during the activation of the Door of Time cutscene.
+ * This is used once the Jewel Actor is done orbiting Link and split up to move into the pedastal slots.
+ */
+void DemoEffect_MoveJewelSplit(PosRot* posRot, DemoEffect* this) {
+    switch (this->jewel.type) {
+        case Demo_Effect_Jewel_Kokiri:
+            posRot->pos.x -= 40.0f;
+            break;
+        case Demo_Effect_Jewel_Goron:
+            break;
+        case Demo_Effect_Jewel_Zora:
+            posRot->pos.x += 40.0f;
+            break;
+        default:
+            break;
+    }
+}
+
+/**
+ * Moves the Jewel Actor spherically from startPos to endPos.
+ * This is used by the Jewel Actor during the Door of Time activation cutscene.
+ * This is run when the Jewels merge from Link and begin orbiting him.
+ */
+void DemoEffect_MoveJewelSpherical(f32 degrees, f32 frameDivisor, Vec3f startPos, Vec3f endPos, f32 radius,
+                                   Vec3s rotation, DemoEffect* this) {
+    s32 pad;
+    s32 pad2;
+    f32 distance;
+    f32 xPos;
+    f32 ySpherical;
+    f32 xzSpherical;
+
+    distance = frameDivisor * sqrtf(SQ(endPos.x - startPos.x) + SQ(endPos.y - startPos.y) + SQ(endPos.z - startPos.z));
+
+    this->actor.posRot.pos.x = radius * cosf(degrees * (M_PI / 180.0f));
+    this->actor.posRot.pos.y = distance;
+    this->actor.posRot.pos.z = radius * sinf(degrees * (M_PI / 180.0f));
+
+    xPos = this->actor.posRot.pos.x;
+    ySpherical = (this->actor.posRot.pos.y * cosf(rotation.x * (M_PI / 32768.0f))) -
```

Comment:
`32768.0f` should be fine as `0x8000`

Reply:
Why `M_PI / 0x8000` as opposed to `M_PI / 32768.0f` or `M_PI / 32768`

Reply:
The current standard has binary angles in hex. Since 32768 here represents a binary angle, 0x8000 makes more sense.